yang/iproute2-tc-qdisc.yang

```diff
@@ -1,221 +1,244 @@
+            }
+            enum "HFSC" {
+                description "";
+            }
+            enum "HTB" {
+                description "";
+            }
+            enum "PRIO" {
+                description "";
+            }
+            enum "QFQ" {
+                description "";
+            }
+        }
+        description "Defines the supported queueing discipline kinds.";
     }
-  }
 
-  container tc-qdiscs {
-    description "Container for all qdisc configurations.";
+    container qdiscs {
+        description "Container for all qdisc configurations.";
+        list qdisc {
+            ipr2cgen:cmd-start;
+            ipr2cgen:cmd-add "tc qdisc add";
+            ipr2cgen:cmd-delete "tc qdisc del";
+            ipr2cgen:cmd-update "tc qdisc replace";
+            ipr2cgen:replace-on-update;
+            key "dev";
+            description "List of qdisc configurations.";
 
-    list qdisc {
-      key "dev qdisc-kind";
-      description "List of qdisc configurations.";
+            leaf dev {
+                type iplink:link-ref;
+                description "Name of device 'interface' the qdisc is attached to.";
+            }
+            leaf parent {
+                type parent-id;
+                description "Location of the qdisc in the traffic control hierarchy.";
+            }
+            container estimator {
+                ipr2cgen:add-static-arg "estimator";
+                description "Configuration for traffic rate estimation.";
+                leaf interval {
+                    type uint32;
+                    description "Time interval in milliseconds for rate estimation.";
+                }
+                leaf ewma-log {
+                    type uint8;
+                    description "Exponential Weighted Moving Average (EWMA) log.
+                    Represents the weight given to older observations in the moving average.";
+                }
+            }
+            container stab {
+                ipr2cgen:add-static-arg "stab";
+                description "Configuration for packet size stabilization.";
+                leaf mtu {
+                    type uint32;
+                    description "Maximum Transmission Unit (MTU) for the stab feature.
+                    Helps in defining the largest packet size that can be handled.";
+                }
+                leaf tsize {
+                    type uint32;
+                    description "Table size for the hash table managing packet sizes.
+                    Affects the granularity of the stabilization.";
+                }
+            }
+            leaf qdisc-kind {
+                ipr2cgen:value-only;
+                type qdisc-kind;
 
-      uses qdisc-config;
+                // ann: don't use leaf name "qdisc-kind" in iproute2 command, only take the value.
+                mandatory true;
+                description "The kind of qdisc to apply.";
+            }
+        }
+        list filter-qdisc {
```

Comment:
filter qdiscs include the following:
**Classful qdiscs**
- htb
- cbq
- prio
- sfq
- dsmark
- ingress

**Classless qdiscs**
- clsact

Not sure how would be best to group them ?

Reply:
Clssct and ingress are special qdiscs that typically used to apply filters and policies directly. Also they don't do any queuing. The parameters for them are different. So i added them in a separate list